spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -73,16 +75,34 @@ public class Mqttv5PahoMessageHandler extends AbstractMqttMessageHandler
 
 	public Mqttv5PahoMessageHandler(String url, String clientId) {
 		super(url, clientId);
-		this.connectionOptions = new MqttConnectionOptions();
-		this.connectionOptions.setServerURIs(new String[]{ url });
-		this.connectionOptions.setAutomaticReconnect(true);
+		this.connectionOptions = buildDefaultConnectionOptions(url);
 	}
 
 	public Mqttv5PahoMessageHandler(MqttConnectionOptions connectionOptions, String clientId) {
 		super(obtainServerUrlFromOptions(connectionOptions), clientId);
 		this.connectionOptions = connectionOptions;
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @since 6.0
+	 */
+	public Mqttv5PahoMessageHandler(ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager) {
+		super(clientManager);
+		this.connectionOptions = clientManager.getConnectionInfo();
+	}
+
+	private static MqttConnectionOptions buildDefaultConnectionOptions(@Nullable String url) {
```

Comment:
I still think that we don't need this method and `url` must not be null for that url-based ctor. 
Exactly the same comment I did for the message-driven channel adapter.

Reply:
Sure. Makes sense.